gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -0,0 +1,246 @@
+        convertedOptions[index++] = option.convert(blobInfo);
+      }
+      return convertedOptions;
+    }
+  }
+
+  /**
+   * Construct a {@code Blob} object for the provided {@code BlobInfo}. The storage service is used
+   * to issue requests.
+   * 
+   * @param storage the storage service used for issuing requests
+   * @param info blob's info
+   */
+  public Blob(Storage storage, BlobInfo info) {
+    this.storage = checkNotNull(storage);
+    this.info = checkNotNull(info);
+  }
+
+  /**
+   * Construct a {@code Blob} object for the provided bucket and blob names. The storage service is
+   * used to issue requests.
+   * 
+   * @param storage the storage service used for issuing requests
+   * @param bucket bucket's name
+   * @param blob blob's name
+   */
+  public Blob(Storage storage, String bucket, String blob) {
+    this.storage = checkNotNull(storage);
+    this.info = BlobInfo.of(bucket, blob);
+  }
+
+  /**
+   * Get the blobs's information.
+   * 
+   * @return a {@code BlobInfo} object for this blob
+   */
+  public BlobInfo info() {
+    return info;
+  }
+
+  /**
+   * Check if this blob exists.
+   *
+   * @return true if this blob exists, false otherwise
+   * @throws StorageException upon failure
+   */
+  public boolean exists() {
+    return storage.get(info.bucket(), info.name()) != null;
+  }
+
+  /**
+   * Return this blob's content.
+   *
+   * @param options blob read options
+   * @return the blob's content
+   * @throws StorageException upon failure
+   */
+  public byte[] content(Storage.BlobSourceOption... options) {
+    return storage.readAllBytes(info.bucket(), info.name(), options);
+  }
+
+  /**
+   * Update the blob's information. Bucket or blob's name cannot be changed by this method. If you
+   * want to rename the blob or move it to a different bucket use the {@link #copyTo} and
+   * {@link #delete} operations.
+   *
+   * @param blobInfo new blob's information. Bucket and blob names must match the current ones
+   * @param options update options
+   * @throws StorageException upon failure
+   */
+  public void update(BlobInfo blobInfo, BlobTargetOption... options) {
+    checkArgument(Objects.equals(blobInfo.bucket(), info.bucket()), "Bucket name must match");
+    checkArgument(Objects.equals(blobInfo.name(), info.name()), "Blob name must match");
+    info = storage.update(blobInfo, options);
+  }
+
+  /**
+   * Delete this blob.
+   *
+   * @param options blob delete options
+   * @return true if blob was deleted
+   * @throws StorageException upon failure
+   */
+  public boolean delete(BlobSourceOption... options) {
+    return storage.delete(info.bucket(), info.name(), convert(info, options));
+  }
+
+  /**
+   * Copy this blob.
+   *
+   * @param target target blob
+   * @param options source blob options
+   * @return the copied blob
+   * @throws StorageException upon failure
+   */
+  public Blob copyTo(BlobInfo target, BlobSourceOption... options) {
```

Comment:
-  **gcloud-go** has a single function `copyObject` that works similarly to our `Datastore.copy`
-  **gcloud-node** has a `File` object with a `copy(target)` method where target can be either the bucket name as string (object name is preserved), a `Bucket` object (object name is preserved) or a `File` object.
-  **gcloud-python** has a `Bucket` object with a `copy_blob` method that takes a source `Blob` object, a destination `Bucket` object and a new name as string
-  **gcloud-ruby** has a `File` object with a `copy` method that takes destination bucket and destination name as strings

I would keep the methods and state:
> The current `Blob` object is only used to retrieve bucket and blob name. This method does not update remote object's metadata other than `name` and `bucket`.

Why do you thing that the second version needs to be simplified? It takes one more parameter wrt the previous that is needed only if you need to set some `BlobTargetOption`. Would you prefer having both parameters wrapped in a `CopyOptions` object?